### DISCUSSION TITLE
Composing Combobox and TagInput together breaks tag navigation

### DISCUSSION THREAD

**Discussion by Jonas-C** [chakra-ui/zag#1626] [General]
# 🐛 Bug report

When composing Combobox and TagInput together, the left and right arrows no longer allows one to navigate between available tags.

## 💥 Steps to reproduce

1. Add some tags.
2. Backspace to highlight the first tag.
3. Press left arrow. Nothing happens.

## 💻 Link to reproduction

https://zag-nextjs-git-changeset-release-main-chakra-ui.vercel.app/composition/combo-tags

## 🧐 Expected behavior

Left and right arrows should function as they usually do.

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Zag Version      | 0.58.3           |
| Browser          |  Firefox          |
| Operating System |  MacOS          |

## 📝 Additional information
Maybe you could expose a function for focusing the next/previous tag?
This also happens in the ArkUI plus example

**Comment by segunadebayo** [comment]
This is by design. If you close the combobox with the Esc key, you can navigate tags afterward

The combobox should consume the keyboard navigation while it's open. Otherwise, it becomes very difficult to manage when we add support for grid navigation in combobox (which need to support the left and right arrow keys)

I love the idea of adding focus methods to the tag's input APIs. Want to open a PR for this?

**Reply by segunadebayo** [reply]
Based on this logic here
https://github.com/chakra-ui/zag/blob/8f71381a68026b8beb22c1107cc4d20b2038723e/packages/machines/tags-input/src/tags-input.connect.ts#L153-L163

You can see that we prevent the arrow navigation for the tags if the input acts as a combobox and it's open.

**Reply by Jonas-C** [reply]
Oh, gotcha! I can open a PR, but I'm not sure when I'll have time. Hopefully this weekend/early next week!

**Reply by Jonas-C** [reply]
Optionally, could we make this into a prop that controls whether the combobox consumes left/right arrows? I'm not too well versed with grid comboxes, but I'd assume that combining a TagInput and a Combobox is a more common scenario?
